src/gui/app.cpp

```diff
@@ -116,10 +116,16 @@ void App::showUi() noexcept
 		win->show();
 	}
 #else
+	setQuitOnLastWindowClosed(false);
 	NeovimQt::MainWindow *win = new NeovimQt::MainWindow(m_connector.get(), opts);
 
+	// delete the main window when closed to emit `destroyed()` signal.
+	win->setAttribute(Qt::WA_DeleteOnClose);
+
 	QObject::connect(instance(), SIGNAL(openFilesTriggered(const QList<QUrl>)),
 		win->shell(), SLOT(openFiles(const QList<QUrl>)));
+	QObject::connect(win, SIGNAL(closing(int)), instance(), SLOT(mainWindowClosing(int)));
```

Comment:
Sorry I didn't notice this earlier; Qt deprecated this syntax.

We should use the following instead:
``` C++
	// Pull#644: Add support for `:cq` return codes. 
	win->setAttribute(Qt::WA_DeleteOnClose);
	setQuitOnLastWindowClosed(false);

	connect(win, &MainWindow::closing, this, &App::mainWindowClosing);
	connect(win, &MainWindow::destroyed, this, &App::exitWithStatus);
```